lib/addons/neuron/index.ts

```diff
@@ -0,0 +1,31 @@
+import { KubernetesManifest } from "aws-cdk-lib/aws-eks";
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import { loadYaml, readYamlDocument } from "../../utils/yaml-utils";
+
+export class NeuronPluginAddOn implements ClusterAddOn {
+    deploy(clusterInfo: ClusterInfo): void {
+        const cluster = clusterInfo.cluster;
+
+        // Read in YAML docs
+        const rbac = readYamlDocument(__dirname + '/k8s-neuron-device-plugin-rbac.ytpl');
+        const plugin = readYamlDocument(__dirname + '/k8s-neuron-device-plugin.ytpl');
+        
+        // Apply Manifests
+        const rbacLoadYaml = rbac.split("---").map(e => loadYaml(e));
+        const rbacManifest = new KubernetesManifest(cluster.stack, "neuron-rbac-manifest", {
```

Comment:
We should not use direct KubernetesManifest for these. The best option is to use the following pattern:

```
const kubectlProvider = new KubectlProvider(clusterInfo);
const statement = kubectlProvider.addManifest(manifestDeployment);
```

The above will enable 1/ consistency and 2/ level of abstraction which can allow storing these manifests in GitOps if needed and 3/ allows passing values if those have replaceable params. 
Examples: search for ` kubectlProvider.addManifest` in the repo

Reply:
addressed.